### ISSUE TITLE
KeyboardInterrupt raised a second time after being caught

### ISSUE THREAD

**Issue by chrysn** [nathan-hoad/gbulb#20]
When an application catches KeyboardInterrupt flying out of a `loop.run_forever()` or similar (typically to initialize orderly shutdown), another KeyboardInterrupt will fly out of any further loop running, in contrast to the Python shipped main loop. Minimal example (comment out the gbulb lines to see it on the asyncio main loop):

```python
import asyncio

import gbulb
gbulb.install()

loop = asyncio.get_event_loop()

try:
    print("Please interrupt with Ctrl-C")
    loop.run_forever()
except KeyboardInterrupt:
    loop.run_until_complete(asyncio.sleep(1))
    print("Done as expected")
```

I managed to work around it by adding a

```python
if hasattr(self._mainloop, "_quit_by_sigint"):
    del self._mainloop._quit_by_sigint
```

before the `self._mainloop.run()` line in gbulb/glib_events.py, but given this touches deep into details of `gi.overrides.GLib.MainLoop`.

Please consider enabling consecutive runs of the main loop after interrupt events.

**Comment by nathan-hoad** [comment]
Oh nice! Good find, and thanks for the proposed fix! There's a pending branch at the moment for Windows support, which I intend to merge in over the weekend - once I've merged that in I'll add your recommended fix.

**Comment by chrysn** [comment]
I wouldn't go as far as put it as a recommended fix -- if the PyGObject decide to change their implementation, the fix would break. Maybe this is something that should be taken into consideration with PyGObject maintainers too?

**Comment by nathan-hoad** [comment]
Sorry for the extreme amount of time passing! I _think_ your fix should mostly be okay? I've opted to go with your fix, and I've added a test - in the event they change it, tests should pick it up. The fix will be a part of 0.6.0.
